ev3dev/core.py

```diff
@@ -2728,20 +2730,36 @@ class InfraredSensor(Sensor, ButtonBase):
             11: ['blue_up', 'blue_down']
             }
 
-    #: Handles ``Red Up`` events.
-    on_red_up = None
-
-    #: Handles ``Red Down`` events.
-    on_red_down = None
-
-    #: Handles ``Blue Up`` events.
-    on_blue_up = None
-
-    #: Handles ``Blue Down`` events.
-    on_blue_down = None
-
-    #: Handles ``Beacon`` events.
-    on_beacon = None
+    _BUTTONS = ('red_up', 'red_down', 'blue_up', 'blue_down', 'beacon')
+
+    # See process() for an explanation on how to use these
+    #: Handles ``Red Up``, etc events on channel 1
+    on_red_up1 = None
```

Comment:
Would `channel1_red_up` (`C1_red_up`, `ch1_red_up`) be easier to understand? Also, would now be a good time to follow up on https://github.com/rhempel/ev3dev-lang-python/pull/374#discussion_r140795846 and change `red_up` to `top_red` or `top_left`?

Reply:
@dlech wouldn't you also need to rename the buttons in the kernel too?

Regarding the on `on_red_up1` I think maybe we look at making these normal methods for the InfraredSensor class and then in the documents show how to make a child class where you override those methods.  To me that would be far easier to follow, the current approach is pretty unique and isn't something most folks have seen.  I like the `channel1_red_up` model for naming.

Reply:
> wouldn't you also need to rename the buttons in the kernel too?

If these names are in the kernel, it is only in the documentation. What you call them in python doesn't depend on what the kernel calls them.

Reply:
ah, I see now. 

changing `red_up` to either `top_left` or `top_red` sounds fine, of the two `top_left` would be my preference.

Reply:
`red_top_pressed`?

Reply:
Having `pressed` in the name would be confusing though because the button may not be pressed. In `red_up` that doesn't mean the button is up, it is referring to the upper button on the red side.

Let's do this, I'll commit a patch with `top_left`, `channel1_top_left`, etc and we can see what it looks like and then go from there.